### PR TITLE
[master] LOG-3447: fix logging-view-plugin deployment to enable features based…

### PR DESCRIPTION
**Pull request by jcantrill**
… upon OCP version

### Description
This is a forward port of https://github.com/openshift/cluster-logging-operator/pull/1819 to ensure the logic makes it into the next release

### Links
* https://issues.redhat.com/browse/LOG-3447

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -199,12 +202,12 @@ func getCLOVersion() (string, error) {
 	return cloversion, nil
 }
 
-// getClusterID retrieves the ID of the cluster
-func getClusterID(k8client client.Reader) (string, error) {
+// getClusterVersion retrieves the ID of the cluster
```
Comment (syedriko) [inline main.go:205]: the version of the cluster

File: internal/visualization/console/feature_map.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+const (
+	featureDevConsole = "dev-console"
+)
+
+var (
+
+	// featuresIfUnmatched represents the default features to enable
+	featuresIfUnmatched = []string{featureDevConsole}
```
Comment (syedriko) [inline internal/visualization/console/feature_map.go:15]: Here an further on there seems to be a mix of terms - matching vs comparison.

File: internal/visualization/console/feature_map_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("#FeaturesForOCP", func() {
+	It("should return an empty set when there is an empty version ", func() {
+		Expect(FeaturesForOCP("")).To(Equal([]string{}))
+	})
+	It("should return the default set when there is no version match ", func() {
```
Comment (syedriko) [inline internal/visualization/console/feature_map_test.go:12]: It's unclear what the version gets matched against.

File: main.go [hunk 4]
```diff
@@ -106,32 +106,35 @@ func main() {
 		os.Exit(1)
 	}
 
-	clusterID, err := getClusterID(mgr.GetAPIReader())
+	clusterVersion, err := getClusterVersion(mgr.GetAPIReader())
```
Comment (syedriko) [inline main.go:109]: This assumes that the cluster version stays the same for the lifetime of a CLO instance, but what
if the cluster is upgraded under a running CLO?

File: test/e2e/consoleplugin/smoke_test/consoleplugin_test.go [hunk 5]
```diff
@@ -76,5 +79,22 @@ var _ = Describe("[ConsolePlugin]", func() {
...
+			}
+			for _, pod := range pods.Items {
+				for _, cond := range pod.Status.Conditions {
+					if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+						return nil
+					}
+				}
+			}
+			return fmt.Errorf("No pods found in ready condition")
+		},
```
Comment (syedriko) [inline test/e2e/consoleplugin/smoke_test/consoleplugin_test.go:97]: Is indentation off here?

File: internal/visualization/console/feature_map.go (hunk 2 above)
Comment (jcantrill) [reply internal/visualization/console/feature_map.go:15]: I don't understand what you are asking or suggesting

File: internal/visualization/console/feature_map_test.go (hunk 3 above)
Comment (jcantrill) [reply internal/visualization/console/feature_map_test.go:12]: is your point it should be called something like "default" or something more verbose then "unamatched"?

File: main.go (hunk 4 above)
Comment (jcantrill) [reply main.go:109]: then the operator will be redeployed.  I don't see how the cluster could be upgraded and the operator not get restarted

File: test/e2e/consoleplugin/smoke_test/consoleplugin_test.go (hunk 5 above)
Comment (jcantrill) [reply test/e2e/consoleplugin/smoke_test/consoleplugin_test.go:97]: :man_shrugging:  I ran it through "make fmt"

**Pull request merged**